File: Backend/Models/AuditLog.cs
using System;

namespace Backend.Models
{
    public class AuditLog
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public string Action { get; set; } = string.Empty;
        public string EntityType { get; set; } = string.Empty;
        public string? EntityId { get; set; }
        public string Summary { get; set; } = string.Empty;
        public AuditLogStatus Status { get; set; } = AuditLogStatus.Success;
        public int ActorUserId { get; set; }
        public string ActorEmail { get; set; } = string.Empty;
        public string? ActorRole { get; set; }
        public string? IpAddress { get; set; }
        public string? UserAgent { get; set; }
        public string? Metadata { get; set; }
        public string? ErrorMessage { get; set; }
        public string? RequestId { get; set; }
        public string? CorrelationId { get; set; }
        public string? SessionId { get; set; }
        public string? TenantId { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: Admin/src/types/auditLog.ts
export type AuditLogListItem = {
  id: string;
  action: string;
  entityType: string;
  entityId?: string | null;
  summary: string;
  status: "Success" | "Failure" | string;
  actorFullName: string;
  ipAddress?: string | null;
  createdAt: string;
};

export type AuditLogDetail = {
  id: string;
  action: string;
  entityType: string;
  entityId?: string | null;
  summary: string;
  status: "Success" | "Failure" | string;
  actorFullName: string;
  actorRole?: string | null;
  ipAddress?: string | null;
  userAgent?: string | null;
  createdAt: string;
  metadata?: Record<string, unknown> | null;
  errorMessage?: string | null;
  requestId?: string | null;
  correlationId?: string | null;
  sessionId?: string | null;
};

export type Filters = {
  from: string;
  to: string;
  userId: string;
  action: string;
  entityType: string;
  entityId: string;
  status: string;
  q: string;
};

export const initialFilters: Filters = {
  from: "",
  to: "",
  userId: "",
  action: "",
  entityType: "",
  entityId: "",
  status: "",
  q: "",
};